mkt/site/storage_utils.py

```diff
@@ -86,6 +86,14 @@ class S3BotoPublicStorage(S3BotoStorage):
         'Cache-Control': 'max-age=31536000',
     }
 
+    def url(self, name, headers=None, response_headers=None):
+        # We want to use the CDN for public ressources instead of directly
+        # calling S3. To do that, we are manually constructing the URL to point
+        # to STATIC_URL (which is our CDN) and we add a "_s3" component to the
+        # URL to help the CDN figure out that the underlying ressource lives on
+        # S3. The bucket is hardcoded in the CDN configuration.
+        return '%s_s3/%s' % (settings.STATIC_URL, name.lstrip('/'))
```

Comment:
I'd prefer to avoid redirects, that would complicate things. I like the `STATIC_URL.startswith(SITE_URL)` idea, I'd also need to check if STATIC_URL starts with `/` but I think it should work.